### ISSUE TITLE
Possible endless loop in mergeComplexNumberFormatMasks

### ISSUE THREAD

**Issue by brainz80** [PHPOffice/PhpSpreadsheet#1792]
This is:

```
- [x] a bug report
- [ ] a feature request
- [ ] **not** a usage question (ask them on https://stackoverflow.com/questions/tagged/phpspreadsheet or https://gitter.im/PHPOffice/PhpSpreadsheet)
```

### What is the expected behavior?
Expected not to get stuck in do-while -loop.

### What is the current behavior?
Gets stuck with e.g.
```php
$numbers = array('434', '22222222222');
$masks = array('ka', ' 0', '00');
```

### Callstack
![image](https://user-images.githubusercontent.com/1257591/104602469-1e5cc480-5684-11eb-82e1-417a2f214aa3.png)

### Which versions of PhpSpreadsheet and PHP are affected?
Tested 1.16.0, 1.15.0 and 1.10.0

### Possible fix
```php
    private static function mergeComplexNumberFormatMasks($numbers, $masks)
    {
        $decimalCount = strlen($numbers[1]);
        $postDecimalMasks = [];

        do {
            $tempMask = array_pop($masks);
            $postDecimalMasks[] = $tempMask;
            $decimalCount -= strlen($tempMask);
        } while ($tempMask !== null && $decimalCount > 0);

        return [
            implode('.', $masks),
            implode('.', array_reverse($postDecimalMasks)),
        ];
    }

**Comment by brainz80** [comment]
Made a pull-request of my fix: https://github.com/PHPOffice/PhpSpreadsheet/pull/1793

**Comment by MarkBaker** [comment]
Fixed in [PR #1793](https://github.com/PHPOffice/PhpSpreadsheet/pull/1793)
